pkg/devfile/parser/data/v2/commands.go

```diff
@@ -65,13 +64,21 @@ func (d *DevfileV2) AddCommands(commands []v1.Command) error {
 }
 
 // UpdateCommand updates the command with the given id
-func (d *DevfileV2) UpdateCommand(command v1.Command) {
+// return an error if the command is not found
+func (d *DevfileV2) UpdateCommand(command v1.Command) error {
+	found := false
 	for i := range d.Commands {
-		if strings.ToLower(d.Commands[i].Id) == strings.ToLower(command.Id) {
+		if d.Commands[i].Id == command.Id {
+			found = true
 			d.Commands[i] = command
-			d.Commands[i].Id = strings.ToLower(d.Commands[i].Id)
+			d.Commands[i].Id = d.Commands[i].Id
+			break
```

Comment:
we can return here if we found the command and updated it

Reply:
updated